lib/segment/src/data_types/vectors.rs

```diff
@@ -15,13 +15,29 @@ use crate::common::utils::transpose_map_into_named_vector;
 use crate::types::{VectorName, VectorNameBuf};
 use crate::vector_storage::query::{ContextQuery, DiscoveryQuery, RecoQuery, TransformInto};
 
+/// How many dimensions of a sparse vector are considered to be a single unit for cost estimation.
+const SPARSE_VECTOR_COST: usize = 64;
+
 #[derive(Clone, Debug, PartialEq)]
 pub enum VectorInternal {
     Dense(DenseVector),
     Sparse(SparseVector),
     MultiDense(MultiDenseVectorInternal),
 }
 
+impl VectorInternal {
+    /// Returns the estimated cost of using this vector in terms of the number of how many similarity comparisons vector will make against one point.
+    pub fn similarity_cost(&self) -> usize {
+        match self {
+            VectorInternal::Dense(_dense) => 1,
+            // TODO(ratelimits): Currently one sparse vector counts as one, regardless of number of dimensions.
+            //                   We should come up with a formula for adjusting its cost.
+            VectorInternal::Sparse(sparse) => 1.max(sparse.indices.len() / SPARSE_VECTOR_COST),
```

Comment:
How about [`div_ceil`](https://doc.rust-lang.org/std/primitive.usize.html#method.div_ceil)?

```suggestion
            VectorInternal::Sparse(sparse) => sparse.indices.len().div_ceil(SPARSE_VECTOR_COST),
```

| Size | Cost max/div | Cost div_ceil | Comment |
|---:|---:|---:|:---|
| 0 | 1 | 0 | Should not be possible, cost 0 seems fine
| 1 | 1 | 1 |
| 64 | 1 | 1 |
| 65 | 1 | 2 | I think 2 is better than 1
| 128 | 2 | 2 |
| 129 | 2 | 3 | I think 3 is better than 2

Reply:
Good idea, thanks